### ISSUE TITLE
Non-replicable results with tf v2.12

### ISSUE THREAD

**Issue by fickas** [keras-team/keras#18244]
I am running tf v2.12. I am concerned about tuning. When I ran simple tuning algorithm on a Sequential model, I came out with different architectures as best on every rerun, i.e., different number of layers and different nodes in each layer. It is kind of hard to average architectures across reruns.

I set both tf and numpy seeds to a constant.

This also affects sklearn tuning algorithms, e.g., RandomSearch and HalvingSearch. They will not produce consistent "best parameters" on rerun. I feel I am missing something because I would expect a hue and cry from the community if tuning libraries stopped working as expected and I have not seen any complaints.

I did just try installing v2.8, and I now get consistent results with RandomSearchCV. When I use 2.12, I do not. So it appears something changed between 2.8 and 2.12. I have not tried 2.9, 2.10, 2.11 to see where the change occurred.

**Comment by fickas** [comment]
It appears the change happened between v2.9 and v2.10. With 2.9, I get replicable results. With 2.10, I do not. I do not see anything in the 2.10 release notes that suggest such a change.

**Comment by tilakrayal** [comment]
@fickas,
I tried to execute the code with the alternative approach by using **tf.keras.utils.set_random_seed**, and I was getting the same result while executing the mentioned code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/d906cd28355958b4ea371cee7ba2866a/different_results.ipynb).

Also it will be suggested to use `tf.keras.utils.set_random_seed` to use the latest versions. It sets all random seeds for the program (Python, NumPy, and TensorFlow). You can use this utility to make almost any Keras program fully deterministic.
https://www.tensorflow.org/api_docs/python/tf/keras/utils/set_random_seed

**tf.random.set_seed** sets the global random seed. Thank you!

**Comment by fickas** [comment]
Thanks. Huge help. Is this documented someplace that I missed?

**Comment by tilakrayal** [comment]
@fickas,
Calling [tf.keras.utils.set_random_seed](https://www.tensorflow.org/api_docs/python/tf/keras/utils/set_random_seed) sets the Python seed, the NumPy seed, and the TensorFlow seed. Setting these seeds is necessary to ensure any random numbers your program generates are also deterministic.

Could you please have a look at this official document for reference.
https://www.tensorflow.org/api_docs/python/tf/config/experimental/enable_op_determinism

Thank you!

**Comment by fickas** [comment]
Again, super helpful, thanks.
